internal/registry/types.builds.go

```diff
@@ -15,3 +17,89 @@ type Build struct {
 	Images        map[string]registryimage.Image
 	Status        models.HashicorpCloudPackerBuildStatus
 }
+
+// NewBuildFromCloudPackerBuild converts a HashicorpCloudePackerBuild to a local build that can be tracked and published to the HCP Packer Registry.
+// Any existing labels or images associated to src will be copied to the returned Build.
+func NewBuildFromCloudPackerBuild(src *models.HashicorpCloudPackerBuild) (*Build, error) {
+
+	build := Build{
+		ID:            src.ID,
+		ComponentType: src.ComponentType,
+		CloudProvider: src.CloudProvider,
+		RunUUID:       src.PackerRunUUID,
+		Status:        src.Status,
+		Labels:        src.Labels,
+	}
+
+	var err error
+	for _, image := range src.Images {
+		image := image
+		err = build.AddImages(registryimage.Image{
+			ImageID:        image.ImageID,
+			ProviderName:   build.CloudProvider,
+			ProviderRegion: image.Region,
+		})
+
+		if err != nil {
+			return nil, fmt.Errorf("NewBuildFromCloudPackerBuild: %w", err)
+		}
+	}
+
+	return &build, nil
+}
+
+// AddLabelsToBuild merges the contents of data to the labels associated with the build.
+// Duplicate keys will be updated to reflect the new value.
+func (b *Build) MergeLabels(data map[string]string) {
+	if data == nil {
+		return
+	}
+
+	if b.Labels == nil {
+		b.Labels = make(map[string]string)
+	}
+
+	for k, v := range data {
+		// TODO: (nywilken) Determine why we skip labels already set
+		//if _, ok := build.Labels[k]; ok {
+		//continue
+		//}
+		b.Labels[k] = v
+	}
+
+}
+
+// AddImages appends one or more images artifacts to the build.
+func (b *Build) AddImages(images ...registryimage.Image) error {
+
+	if b.Images == nil {
+		b.Images = make(map[string]registryimage.Image)
+	}
+
+	for _, image := range images {
+		image := image
+
+		if err := image.Validate(); err != nil {
+			return fmt.Errorf("AddImages: failed to add image to build %q: %w", b.ComponentType, err)
+		}
+
+		if b.CloudProvider == "" {
+			b.CloudProvider = image.ProviderName
+		}
+
+		b.MergeLabels(image.Labels)
+		b.Images[image.String()] = image
+	}
+
+	return nil
+}
+
+// IsNotDone returns true if build does not satisfy all requirements of a completed build.
+// A completed build must have a valid ID, one or more Images, and its Status is HashicorpCloudPackerBuildStatusDONE.
+func (b *Build) IsNotDone() bool {
+	hasBuildID := b.ID != ""
+	hasImages := len(b.Images) == 0
+	isNotDone := b.Status != models.HashicorpCloudPackerBuildStatusDONE
```

Comment:
Would this validation be enough?

Reply:
I believe so. Do you think we should validate something else? 


The rationale is as follows: 

1. A build without and Id has not been registered on the registry therefore it doesn't exist and not worth considering.
2. A build that exists on the registry with no images is by the service specs not done. 
3. An image can only be set to the done status if it has at least one image.

Reply:
Sorry, I confused you. I was saying that this validation `isNotDone := b.Status != models.HashicorpCloudPackerBuildStatusDONE` I expected to be enough 🤔
I'm wanted to know if you had any other use cases that this wasn't enough.

Reply:
Ah got it. This might be artifact of an edge case earlier on where we may have created or put a build in a done status without having images.But I can't recall immediately. I believe the API does not allow a build to be put in the DONE status without images so that check might not be needed 

I have to review previous commits and notes before looking to remove the other conditions 😄